s4e-web/src/app/utils/error-interceptor/error.interceptor.ts

```diff
@@ -1,24 +1,99 @@
+import { catchError, filter, pairwise, map, last } from 'rxjs/operators';
+import { Router, RoutesRecognized } from '@angular/router';
+import {
+  ERROR_CODE_FORBIDDEN,
+  ERROR_CODE_UNAUTHORIZED,
+  ERROR_CODE_TIMEOUT,
+  ERROR_CODE_BAD_REQUEST,
+  ERROR_CODE_NOT_FOUND,
+  ERROR_CODE_BAD_GATEWAY,
+  ERROR_CODE_INTERNAL_SERVER_ERROR
+} from 'src/app/errors/errors.model';
+
+import * as _ from 'lodash';
+
+export const ERROR_INTERCEPTOR_SKIP_HEADER = 'Error-Interceptor-Skip-Header';
+
+class HttpErrorHelper {
+  public static isServerSideError = (error: HttpErrorResponse): boolean => error instanceof HttpErrorResponse;
+
+  public static isClientSideError = (error: HttpErrorResponse): boolean => error instanceof HttpErrorResponse
+    && error.error instanceof ErrorEvent
+
+  public static hasSkipHeader = (request: HttpRequest<any>): boolean => request.headers.has(ERROR_INTERCEPTOR_SKIP_HEADER);
+
+  public static skippedHttpEvent = (request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> => next
+    .handle(request.clone({ headers: request.headers.delete(ERROR_INTERCEPTOR_SKIP_HEADER) }))
+}
 
 @Injectable({
   providedIn: 'root',
 })
 export class ErrorInterceptor implements HttpInterceptor {
+  private _backLink: string;
+
+  constructor(private _router: Router) {
+    this._router.events
+      .pipe(
+        filter(event => event instanceof RoutesRecognized),
+        pairwise(),
+        map((event: [RoutesRecognized, RoutesRecognized]) => event[0].url)
+      )
+      .subscribe((lastUrl: string) => this._backLink = lastUrl);
+  }
+
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    return next.handle(request).pipe(catchError((error: HttpErrorResponse) => {
-      // switch(error.status) {
-      //   case 400: {
-      //     break;
-      //   }
-      //   default: {
-      //
-      //   }
-      // }
-      // const error = (error.error && error.error.message) || error.statusText;
-      return throwError(error);
-    }));
+    return next.handle(request)
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          if (HttpErrorHelper.hasSkipHeader(request)) {
+            return HttpErrorHelper.skippedHttpEvent(request, next);
+          }
+
+          if (HttpErrorHelper.isClientSideError(error)) {
+            console.error(error.message);
+            return throwError(error);
+          }
+
+          if (HttpErrorHelper.isServerSideError(error)) {
+            this._handleHttpErrorResponses(error);
+          }
+
+          return throwError(error);
+        })
+      );
+  }
+
+  private _handleHttpErrorResponses = (error: HttpErrorResponse) => {
+    console.log(`error status : ${error.status} ${error.statusText}`);
+    switch (error.status) {
+      case ERROR_CODE_FORBIDDEN:
+      case ERROR_CODE_UNAUTHORIZED:
+        this._router.navigate(
+          ['/login'],
+          {
+            queryParams: {backLink: this._backLink}
```

Comment:
when reading and creating back links later you add `/` prefix. I think it would be better to add it here.